File: todo33/spec.rb
require './match_values'
# Get keys of a hash whose values equal to given arguments.

describe Hash do
  describe "#keys_of" do
 
    it "should return keys of the given value when there's one answer" do
      expect({a: 1, b: 2, c: 3}.keys_of(1)).to eq([:a])
    end

    it "should work when there's two answers" do
      expect({a: 1, b: 2, c: 3, d: 1}.keys_of(1)).to eq([:a, :d])
    end

    it "should work when there's two arguments" do
      expect({a: 1, b: 2, c: 3, d: 1}.keys_of(1, 2)).to eq([:a, :b, :d])
    end

  end
end

File: todo42/Jasmine1.3.1/spec/spec.js
// prereqs: arrays, methods, conditional logic
// Write a method that returns the first even element in a array of integers
// This is an old todo we did in Ruby, please rewrite your answer in javascript

describe( "#firstEven", function() {
  it("should return the first even when there is one even", function() {
    expect(firstEven([3, 4, 5])).toBe(4);
  });      
  it("should return the first even when there is more than one even", function() {
    expect(firstEven([3, 5, 6, 7, 2])).toBe(6);
  }); 
  it("should return undefined when there are no evens", function() {
    expect(firstEven[3, 5, 7]).toBe(undefined);
  }); 
});
